File: 2024/albert-arvidsson-polyglot/day_03/part_1/solve.mjs
#!/usr/bin/env node
import { readFileSync } from "node:fs";

const input = readFileSync(process.argv[2], "utf-8");
const regex = /mul\((\d{1,3}),(\d{1,3})\)/g;
let sum = 0;
for (const [, x, y] of input.matchAll(regex)) sum += +x * +y;
console.log(sum);

File: 2024/albert-arvidsson-polyglot/day_03/part_1/solve.exs
#!/usr/bin/env elixir

~r/mul\((\d{1,3}),(\d{1,3})\)/
|> Regex.scan(System.argv() |> hd |> File.read!(), capture: :all_but_first)
|> Enum.map(fn [x, y] -> String.to_integer(x) * String.to_integer(y) end)
|> Enum.sum
|> IO.inspect()
